libraries/abstractions/common_io/include/iot_sdio.h

```diff
@@ -0,0 +1,709 @@
+ *
+ * @design consideration
+ *          SDIO HAL APIs performs protocol layer tasks between sdio host and sdio card.
+ *          Some APIs perform a single task and some APIs perform a sequence of tasks.
+ *          For examples, iot_sdio_io_read_direct() API performs a single task of reading
+ *          one byte from card register, and iot_sdio_card_connect() API performs all tasks
+ *          required by card initialization as specified in section 3.1.2 in "SD Specification
+ *          Part E1 SDIO".
+ *
+ *          Upon calling iot_sdio_open() API, the user is a given a pxSdioHandle, an abstract
+ *          reference to the SDIO card slot that the user is interested in. Thereafter, user
+ *          uses this handle to call other APIs to perform tasks associated with the target
+ *          card slot. For example, iot_sdio_check_card_presence_status(pxSdioHandle, true)
+ *          checks sdio card presence, and iot_sdio_io_write_extended(pxSdioHandle)
+ *          writes data to sdio card in the slot, etc.
+ *
+ * @limitation
+ *          SDIO HAL APIs are intended to be used in embedded systems where each sdio host
+ *          controller is connected to either a single hot swappable sido card slot or multiple
+ *          eSDIO devices in a Shared Bus Configuration (Refer to Section 7.3 of "SD Specification
+ *          Part E1 SDIO").
+ *
+ *          These APIs are not suitable for systems where a single sdio host controller supports
+ *          multiple hot swappable slots. With multiple hot swappable slots available on a single
+ *          host controller, it is not possible pre-determine which card slot sdio device will be
+ *          inserted, and know which card slot handle to request. Since this type of host controller
+ *          is very rare even in non-embedded world, it is not worth to add complexity to SDIO HAL
+ *          APIs to support it.
+ */
+
+#ifndef _IOT_SDIO_H_
+#define _IOT_SDIO_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+
+/**
+ * @defgroup iot_sdio SDIO HAL APIs
+ * @{
+ */
+
+/**
+ * @brief The return codes for the functions in SDIO
+ */
+ /*!< SDIO operation completed successfully. */
+#define IOT_SDIO_SUCCESS                             ( 0 )
+/*!< SDIO ioctl failed. */
+#define IOT_SDIO_IOCTL_FAIL                          ( 1 )
+/*!< SDIO host controller initialization failed. */
+#define IOT_SDIO_HOST_INIT_FAIL                      ( 2 )
+/*!< At least one parameter is invalid. */
+#define IOT_SDIO_INVALID_VALUE                       ( 3 )
+/*!< Sending command to or receive response from
+     card failed. */
+#define IOT_SDIO_TRANSFER_FAIL                       ( 4 )
+/*!< If the card is initialized in 3.3V signaling,
+     and card support high speed mode (SHS ==1),
+     host can issue CMD52 in RAW mode, setting EHS
+     bit in CCCR to one to switch to high speed mode
+     . This error code indicates that the sdio card
```

Comment:
Period should be on previous line. More generally, comments could instead be justified to 80 columns.

Reply:
adjusted.